### PR TITLE
add rootTocPath for version

### PR DESCRIPTION
**Pull request by superyyrrzz**
#818 
@chenkennt @vwxyzh @hellosnow @ansyral @qinezh

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.Engine/SystemMetadataGenerator.cs [hunk 1]
```diff
@@ -48,24 +48,26 @@ public SystemMetadata Generate(InternalManifestItem item)
...
+                if (((TypeForwardedToRelativePath)_context.RootTocPath).IsFromWorkingFolder())
                 {
+                    attrs.RootTocKey = _context.RootTocPath;
+                    var rootTocPath = ((TypeForwardedToRelativePath)_context.RootTocPath).RemoveWorkingFolder();
                     attrs.RootTocPath = rootTocPath;
-                    var rootTocRelativePath = rootTocPath.MakeRelativeTo(file);
-                    attrs.RelativePathToRootToc = rootTocRelativePath;
-                    attrs.RootTocKey = rootToc.Key;
-                    Logger.LogVerbose($"Root TOC file {rootTocPath} is found.");
+                    attrs.RelativePathToRootToc = rootTocPath.MakeRelativeTo(file);
```
Comment (ansyral) [inline src/Microsoft.DocAsCode.Build.Engine/SystemMetadataGenerator.cs]: >rootTocPath.MakeRelativeTo(file) [](start = 50, length = 32)

file.MakeRelativeTo(rootTocPath)?

File: src/Microsoft.DocAsCode.Build.Engine/SystemMetadataGenerator.cs [hunk 2]
```diff
@@ -48,24 +48,26 @@ public SystemMetadata Generate(InternalManifestItem item)
...
 
-            // 1. Root Toc is always in the top directory of output folder
-            var rootToc = _toc.FirstOrDefault();
-            if (rootToc != null)
+            // 1. Root Toc is specified by RootTocKey, or by default in the top directory of output folder
+            if (!string.IsNullOrEmpty(_context.RootTocPath))
             {
-                var rootTocPath = rootToc.File.RemoveWorkingFolder();
-                if (rootTocPath.SubdirectoryCount == 0)
+                if (((TypeForwardedToRelativePath)_context.RootTocPath).IsFromWorkingFolder())
```
Comment (ansyral) [inline src/Microsoft.DocAsCode.Build.Engine/SystemMetadataGenerator.cs]: >((TypeForwardedToRelativePath)_context.RootTocPath).IsFromWorkingFolder()) [](start = 20, length = 74)

by default paths in docfx.json are from working folder, does user need to specify `~` for roottocpath?

Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Build.Engine/SystemMetadataGenerator.cs]: Seems no need. removed.

---
In reply to: [89045892](https://github.com/dotnet/docfx/pull/908#discussion_r89045892) [](ancestors = 89045892)

**Comment by ansyral** [comment]
:shipit:

**Comment by qinezh** [comment]
:shipit:

**Pull request merged**